content/web-api-rest/reading/what-is-rest/_index.md

```diff
@@ -0,0 +1,179 @@
+that adheres to it. 
+
+The same benefits are experienced by the API consumer. REST is a set of guiding principles for supporting 
+the organization of an API's core responsibilities -- managing and transferring data. The REST pattern organizes the external 
+interface, or contract, and does not concern itself with the internal implementation of the API.
+
+Adopting the REST specification into the design of an API provides consistency during development and consumption. Much like 
+following the patterns of MVC allows other developers to easily understand your code, following REST patterns gives other developers 
+the benefit of understanding how your API is structured and behaves. As an added bonus, a REST API also gives the client application 
+a base-line understanding on how to interact with your API.
+
+{{% notice green "Tip" "rocket" %}}
+The topics of state and representation are *purposefully abstract* in REST so that they can be applied to any API. Don't get overwhelmed!
+{{% /notice %}}
+   
+### What is State?
+
+State is transitional application data that can be viewed or changed by external interaction. Though state is abstract, programmers interact with it using CRUD operations. 
+
+Imagine viewing, or reading, the state of an application's data as it transitions through each of the following interactions:
+
+1. **Before creating**: Empty state
+2. **After creating**: Initial state
+3. **After updating**: New state
+4. **After deleting**: Empty state
+
+You can see that the state is defined by how the data exists after its latest interaction. 
+
+{{% notice blue "Note" "rocket" %}}
+The concept of state is both the most abstract and most fundamental aspect of REST. 
+{{% /notice %}}
+
+### What is a Representation?
+
+Representation refers to a depiction of state that is usable in a given context. In the REST context, state must be 
+represented in a way that is portable and compatible with both the client and API. 
+
+JSON is a data format that provides structure, portability and compatibility. For these reasons, JSON is the standard representation used when transferring state between a client application and an API. 
+
+{{% notice blue "Example" "rocket" %}}
+The state of a *single* ``CodingEvent`` entity is represented as a single JSON object:
+{{% /notice %}}
+
+```json {linenos=table}
+{
+    "Id": 1,
+    "Title": "Halloween Hackathon!",
+    "Description": "A gathering of nerdy ghouls...",
+    "Date": "2020-10-31"
+}
+```
+
+Whereas the state of a collection of `CodingEvents` is represented by a JSON array of objects.
+
+```json {linenos=table}
+[
+    {
+    "Id": 1,
+    "Title": "Halloween Hackathon!",
+    "Description": "A gathering of nerdy ghouls...",
+    "Date": "2020-10-31"
+    },
+    ...
+]
+```
+
+Notice that the state here is represented as the collective state of all of the ``CodingEvents`` in the collection.
```

Comment:
```suggestion
{{% notice green "Example" "rocket" %}}
The state of a *single* ``CodingEvent`` entity is represented as a single JSON object:

```json {linenos=table}
{
    "Id": 1,
    "Title": "Halloween Hackathon!",
    "Description": "A gathering of nerdy ghouls...",
    "Date": "2020-10-31"
}
```

Whereas the state of a collection of `CodingEvents` is represented by a JSON array of objects.

```json {linenos=table}
[
    {
    "Id": 1,
    "Title": "Halloween Hackathon!",
    "Description": "A gathering of nerdy ghouls...",
    "Date": "2020-10-31"
    },
    ...
]
```

Notice that the state here is represented as the collective state of all of the ``CodingEvents`` in the collection.
{{% /notice %}}
```